### PR TITLE
Fix(Core) Fix many errors caused by updating smartscripts

### PR DESCRIPTION
**Pull request by pklloveyou**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/AI/SmartScripts/SmartScript.cpp [hunk 1]
```diff
@@ -847,23 +855,28 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
         }
         case SMART_ACTION_CALL_GROUPEVENTHAPPENS:
         {
-            if (!unit)
```
Comment (Nefertumm) [inline src/server/game/AI/SmartScripts/SmartScript.cpp:850]: Restore this check, you are dereferencing unit below, which could lead to a segmentation fault crash.

Comment (pklloveyou) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:850]: I'll try it with it

Comment (pklloveyou) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:850]: After that, task 12687 cannot be completed

Comment (Nefertumm) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:850]: I don't understand, but don't remove that check or we will have a crash later. If there is an issue with something else, then is an issue with the script itself and not with the SAI system

Comment (pklloveyou) [reply src/server/game/AI/SmartScripts/SmartScript.cpp:850]: ok

**Comment by pklloveyou** [comment]
I deleted some custom modifications Please check again

**Pull request closed without merging**